### PR TITLE
Add last "description" meta tags

### LINKED ISSUE #5767: Add <meta name="description”> for all other pages
Following on from https://github.com/mozilla/addons-frontend/issues/5766 we should also add descriptions to all other relevant pages represented by the new frontend.

See also https://docs.google.com/document/d/1W9LOw7wgbpY6Mw5Zd4V71hJ0YahY4DTNl23MHVpG0_E/edit?ts=5b5b40e5#heading=h.aqjhycqt8ulj

## Pages

⚠️ all of the page below are available for Firefox and Android (Firefox for Android):

- [x] Homepage: https://addons.mozilla.org/en-US/firefox/
- [x] Landing Extensions: https://addons.mozilla.org/en-US/firefox/extensions/
- [x] Landing Themes: https://addons.mozilla.org/en-US/firefox/themes/
- [x] Search tools: https://addons.mozilla.org/en-US/firefox/search-tools/
- [x] Reviews for an add-on: https://addons.mozilla.org/en-US/firefox/addon/user-agent-string-switcher/reviews/ --> https://github.com/mozilla/addons-frontend/pull/6712
- [x] user profile of a developer or artist or both: https://addons.mozilla.org/en-US/firefox/user/Testfahrer/
- Themes / category pages (https://addons.mozilla.org/en-US/firefox/themes/):
  - [x] Abstract
  - [x] Causes
  - [x] Fashion
  - [x] Film and TV
  - [x] Firefox
  - [x] Foxkeh
  - [x] Holiday
  - [x] Music
  - [x] Nature
  - [x] Other
  - [x] Scenery
  - [x] Seasonal
  - [x] Solid
  - [x] Sports
  - [x] Websites
- [x] About page: https://addons.mozilla.org/en-US/about
- [x] Review guide: https://addons.mozilla.org/en-US/review_guide
- [x] Collection pages (we can probably rely on the description of the collection, but there might be something to add before): https://addons.mozilla.org/en-US/firefox/collections/mozilla/ad-blockers/

Firefox-only pages:

- [x] Dictionaries and Lang Packs: https://addons.mozilla.org/en-US/firefox/language-tools/
- Extensions / category pages (https://addons.mozilla.org/en-US/firefox/extensions/):
  - [x] Alerts & Updates
  - [x] Appearance
  - [x] Bookmarks
  - [x] Download Management
  - [x] Feeds, News & Blogging
  - [x] Games & Entertainment
  - [x] Language Support
  - [x] Other
  - [x] Photos, Music & Videos
  - [x] Privacy & Security
  - [x] Search Tools
  - [x] Shopping
  - [x] Social & Communication
  - [x] Tabs
  - [x] Web Development

Android-only pages:

- Extensions / category pages (https://addons.mozilla.org/en-US/android/extensions/):
  - [x] Device Features & Location
  - [x] Experimental
  - [x] Feeds, News, & Blogging
  - [x] Other
  - [x] Performance
  - [x] Photos & Media
  - [x] Security & Privacy
  - [x] Shopping
  - [x] Social Networking
  - [x] Sports & Games
  - [x] User Interface

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #5767

---

- [x] Review guide: addons.mozilla.org/en-US/review_guide
- [x] Collection pages
- [x] User profile pages
- [x] also update some other descriptions to their latest revision

### CODE REVIEW

File: src/amo/pages/Collection/index.js [hunk 1]
```diff
@@ -506,6 +506,22 @@ export class CollectionBase extends React.Component<InternalProps> {
...
 
+  getPageDescription() {
+    const { collection, i18n } = this.props;
+
+    invariant(collection, 'collection is required');
+
+    return i18n.sprintf(
+      i18n.gettext(`Firefox collections make it easy to keep track of favorite
+        extensions and themes, and share your perfectly customized browser with
+        others. Explore the %(collectionName)s: %(collectionDescription)s.`),
```
Comment (rebmullin) [inline src/amo/pages/Collection/index.js]: it seems like description is optional so it looks a little odd if it's missing. See here:

Alt text

can we show this without the `:` when there is no description available?

Also, just wondering on the wording here - would it sound better if it says "Explore the [CollectionName] Collection."  or "Explore [CollectionName]"? I bet this was already decided and covered somewhere but just double checking : )

File: tests/unit/amo/pages/TestUserProfile.js [hunk 2]
```diff
@@ -965,6 +965,88 @@ describe(__filename, () => {
...
+      }),
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('meta[name="description"]')).toHaveLength(1);
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(`The profile of ${displayName}, an extension author`),
+    );
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(`by ${displayName}`),
```
Comment (rebmullin) [inline tests/unit/amo/pages/TestUserProfile.js]: do we need both checks here?

File: tests/unit/amo/pages/TestUserProfile.js [hunk 3]
```diff
@@ -965,6 +965,88 @@ describe(__filename, () => {
...
+      }),
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('meta[name="description"]')).toHaveLength(1);
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(`The profile of ${displayName}, a theme author`),
+    );
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(`by ${displayName}`),
```
Comment (rebmullin) [inline tests/unit/amo/pages/TestUserProfile.js]: same here...

File: tests/unit/amo/pages/TestUserProfile.js [hunk 4]
```diff
@@ -965,6 +965,88 @@ describe(__filename, () => {
...
+    const root = renderUserProfile({ store });
+
+    expect(root.find('meta[name="description"]')).toHaveLength(1);
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(
+        `The profile of ${displayName}, an extension and theme author`,
+      ),
+    );
+    expect(root.find('meta[name="description"]').prop('content')).toMatch(
+      new RegExp(`by ${displayName}`),
```
Comment (rebmullin) [inline tests/unit/amo/pages/TestUserProfile.js]: same here...

File: tests/unit/amo/pages/TestUserProfile.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/amo/pages/TestUserProfile.js]: yep, to make sure the i18n var is correctly replaced.

File: tests/unit/amo/pages/TestUserProfile.js (hunk 3 above)
Comment (willdurand) [reply tests/unit/amo/pages/TestUserProfile.js]: yep, to make sure the i18n var is correctly replaced.

File: tests/unit/amo/pages/TestUserProfile.js (hunk 4 above)
Comment (willdurand) [reply tests/unit/amo/pages/TestUserProfile.js]: yep, to make sure the i18n var is correctly replaced.

File: src/amo/pages/Collection/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/pages/Collection/index.js]: All Mozilla collections have a description (or should). Only Mozilla collections are indexed, so I guess it is fine to leave it as is for now.

**Comment by willdurand** [comment]
> Also, I didn't see any meta description for the Collection landing page. Will we be doing anything for that? It wasn't clear to me...

Which collection "landing page"? The "description" meta tag is for (detail) collection pages only.

File: src/amo/pages/Collection/index.js (hunk 1 above)
Comment (willdurand) [reply src/amo/pages/Collection/index.js]: That being said, the latest revision makes it easier to handle both cases, so I did it :)

Comment (rebmullin) [reply src/amo/pages/Collection/index.js]: seems fine to me : )

**Comment by willdurand** [comment]
Thanks!

**Pull request merged**